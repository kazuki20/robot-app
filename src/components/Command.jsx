import React from "react";
import { Row, Col, Button } from "reactstrap";
import compass from "../img/compass.png";

export default function Command({
  report,
  move,
  left,
  right,
  isRobotPlaced,
  toggleModalPlaceInput
}) {
  return (
    <div
      style={{
        height: "90%",
        border: "2px solid",
        textAlign: "center"
      }}
    >
      <Row className="mb-4 mt-4">
        <Col>
          <img src={compass} alt="Compass" style={{height: "20vh"}}/>
        </Col>
      </Row>
      <Row className="mb-4 mt-4">
        <Col>
          <Button
            color="primary"
            className="mr-2"
            onClick={toggleModalPlaceInput}
            name="place"
          >
            Place
          </Button>
          <Button
            color="primary"
            className="ml-2"
            onClick={report}
            disabled={isRobotPlaced ? false : true}
            name="report"
          >
            Report
          </Button>
        </Col>
      </Row>
      <Row>
        <Col>
          <Button
            color="primary"
            onClick={move}
            disabled={isRobotPlaced ? false : true}
            name="move"
          >
            Move ü†ù
          </Button>
        </Col>
      </Row>
      <Row>
        <Col>
          <Button
            color="primary"
            onClick={left}
            disabled={isRobotPlaced ? false : true}
            name="left"
          >
            ‚ü≤ Left
          </Button>
        </Col>
        <Col>
          <Button
            color="primary"
            onClick={right}
            disabled={isRobotPlaced ? false : true}
            name="right"
          >
            Right ‚ü≥
          </Button>
        </Col>
      </Row>
    </div>
  );
}
