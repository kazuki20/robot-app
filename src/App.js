import React from 'react';
import './App.css';
import Table from './components/Table'
import Command from './components/Command'
import {Modal, ModalHeader, ModalBody} from 'reactstrap';

export default class App extends React.Component {
	constructor(props) {
		super(props);

		this.state = {
			robotPosition: {
				x: 0,
				y: 0,
				direction: "NORTH",
			},
			directions: ["NORTH", "EAST", "SOUTH", "WEST"],
			isRobotPlaced: false,
			showModalPlaceInput: false
		}
	}

	move = () => {
		const currentDirection = this.state.robotPosition.direction;

		if(currentDirection === "NORTH") {

			if(this.state.robotPosition.y === 0) {
				return alert("Yay! You are at the edge of the table.");
			}

			this.setState(prevState => ({
				robotPosition: {
					...prevState.robotPosition,
					y: this.state.robotPosition.y-1
				}
			}));
		}

		if(currentDirection === "EAST") {

			if(this.state.robotPosition.x === 4) {
				return alert("Yay! You are at the edge of the table.");
			}

			this.setState(prevState => ({
				robotPosition: {
					...prevState.robotPosition,
					x: this.state.robotPosition.x+1
				}
			}));
		}

		if(currentDirection === "SOUTH") {

			if(this.state.robotPosition.y === 4) {
				return alert("Yay! You are at the edge of the table.");
			}

			this.setState(prevState => ({
				robotPosition: {
					...prevState.robotPosition,
					y: this.state.robotPosition.y+1
				}
			}));
		}

		if(currentDirection === "WEST") {

			if(this.state.robotPosition.x === 0) {
				return alert("Yay! You are at the edge of the table.");
			}

			this.setState(prevState => ({
				robotPosition: {
					...prevState.robotPosition,
					x: this.state.robotPosition.x-1
				}
			}));
		}
	}

	left = () => {
		const currentDirection = this.state.robotPosition.direction;
		var index = this.state.directions.indexOf(currentDirection);

		if(index >= 1) {
			index -= 1;
		} else {
			index = 3;
		}

		this.setRobotDirection(index);
	}

	right = () => {
		const currentDirection = this.state.robotPosition.direction;
		var index = this.state.directions.indexOf(currentDirection);

		if(index < (this.state.directions.length-1)) {
			index += 1;
		} else {
			index = 0;
		}

		this.setRobotDirection(index);
	}

	setRobotDirection = (index) => {
		this.setState(prevState => ({
			robotPosition: {
				...prevState.robotPosition,
				direction: this.state.directions[index]
			}
		}));
	}

	render() {
		console.log(this.state.robotPosition)
		return (
			<div className="App">
		      <Table robotPosition={this.state.robotPosition} />
		      <Command 
		      	right={this.right} 
		      	left={this.left}
		      	move={this.move}
		      	isRobotPlaced={this.state.isRobotPlaced}
		      />
		    </div>
		);
	}
}

