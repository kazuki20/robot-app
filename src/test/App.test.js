import React from 'react';
import { shallow } from 'enzyme';
import Command from '../components/Command';
import App from '../App';

let wrapper;

beforeEach(() => {
    wrapper = shallow(<App />);
});

describe('<App /> rendering', () => {
    it('renders correctly', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('should render one <Command>', () => {
        expect(wrapper.find(Command)).toHaveLength(1);
    });
});