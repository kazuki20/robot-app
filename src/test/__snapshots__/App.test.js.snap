// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> rendering renders correctly 1`] = `
<div
  className="App"
>
  <Table
    robotPosition={
      Object {
        "direction": "NORTH",
        "positionX": 0,
        "positionY": 0,
      }
    }
  />
  <Command
    isRobotPlaced={false}
    left={[Function]}
    move={[Function]}
    report={[Function]}
    right={[Function]}
    toggleModalPlaceInput={[Function]}
  />
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    centered={false}
    fade={true}
    isOpen={false}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    returnFocusAfterClose={true}
    role="dialog"
    scrollable={false}
    toggle={[Function]}
    unmountOnClose={true}
    zIndex={1050}
  >
    <ModalHeader
      charCode={215}
      closeAriaLabel="Close"
      tag="h5"
      wrapTag="div"
    >
      Place Robot Position
    </ModalHeader>
    <Form
      onSubmit={[Function]}
      tag="form"
    >
      <ModalBody
        tag="div"
      >
        <Row
          form={true}
          tag="div"
        >
          <Col
            md={3}
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <FormGroup
              tag="div"
            >
              <Label
                for="exampleCity"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                Position X
              </Label>
              <Input
                name="positionX"
                onChange={[Function]}
                type="select"
                value={0}
              >
                <option>
                  0
                </option>
                <option>
                  1
                </option>
                <option>
                  2
                </option>
                <option>
                  3
                </option>
                <option>
                  4
                </option>
              </Input>
            </FormGroup>
          </Col>
          <Col
            md={3}
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <FormGroup
              tag="div"
            >
              <Label
                for="exampleState"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                Position Y
              </Label>
              <Input
                name="positionY"
                onChange={[Function]}
                type="select"
                value={0}
              >
                <option>
                  0
                </option>
                <option>
                  1
                </option>
                <option>
                  2
                </option>
                <option>
                  4
                </option>
              </Input>
            </FormGroup>
          </Col>
          <Col
            md={6}
            tag="div"
            widths={
              Array [
                "xs",
                "sm",
                "md",
                "lg",
                "xl",
              ]
            }
          >
            <FormGroup
              tag="div"
            >
              <Label
                for="exampleZip"
                tag="label"
                widths={
                  Array [
                    "xs",
                    "sm",
                    "md",
                    "lg",
                    "xl",
                  ]
                }
              >
                Direction
              </Label>
              <Input
                name="direction"
                onChange={[Function]}
                type="select"
                value="NORTH"
              >
                <option
                  key="0"
                >
                  NORTH
                </option>
                <option
                  key="1"
                >
                  EAST
                </option>
                <option
                  key="2"
                >
                  SOUTH
                </option>
                <option
                  key="3"
                >
                  WEST
                </option>
              </Input>
            </FormGroup>
          </Col>
        </Row>
      </ModalBody>
      <ModalFooter
        tag="div"
      >
        <Button
          color="secondary"
          onClick={[Function]}
          tag="button"
        >
          Cancel
        </Button>
        <Button
          color="primary"
          tag="button"
          type="submit"
        >
          Place Robot
        </Button>
         
      </ModalFooter>
    </Form>
  </Modal>
  <Modal
    autoFocus={true}
    backdrop={true}
    backdropTransition={
      Object {
        "mountOnEnter": true,
        "timeout": 150,
      }
    }
    centered={false}
    fade={true}
    isOpen={false}
    keyboard={true}
    modalTransition={
      Object {
        "timeout": 300,
      }
    }
    onClosed={[Function]}
    onOpened={[Function]}
    returnFocusAfterClose={true}
    role="dialog"
    scrollable={false}
    toggle={[Function]}
    unmountOnClose={true}
    zIndex={1050}
  >
    <ModalHeader
      charCode={215}
      closeAriaLabel="Close"
      tag="h5"
      wrapTag="div"
    >
      Robot Position Report
    </ModalHeader>
    <ModalBody
      tag="div"
    >
      Position X is 
      0
       
      <br />
      Position Y is 
      0
       
      <br />
      Facing in 
      NORTH
       
      <br />
    </ModalBody>
    <ModalFooter
      tag="div"
    >
      <Button
        color="secondary"
        onClick={[Function]}
        tag="button"
      >
        Ok
      </Button>
    </ModalFooter>
  </Modal>
</div>
`;
